torchvision/datasets/video_utils.py

```diff
@@ -255,6 +255,50 @@ def get_clip_location(self, idx):
             clip_idx = idx - self.cumulative_sizes[video_idx - 1]
         return video_idx, clip_idx
 
+    def get_idx_range_of_video(self, idx):
+        """
+        Provided an idx of a clip, the function returns the indices range belong to the same video
+        """
+
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        if video_idx == 0:
+            idx_start = 0
+        else:
+            idx_start = self.cumulative_sizes[video_idx - 1]
+
+        if len(self.cumulative_sizes) == 0:
+            idx_end = 0
+        else:
+            idx_end = self.cumulative_sizes[video_idx] - 1
+
+        return idx_start, idx_end
+
+    def get_video_path(self, idx):
+        """
+        Converts a flattened representation of the indices into a video path
+        """
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        return self.video_paths[video_idx]
+
+    def get_video_name(self, idx):
+        """
+        Converts a flattened representation of the indices into a video name
+        """
+        from os import path, sep
+        video_path = self.get_video_path(idx)
+        file_no_ext, ext = path.splitext(video_path)
+        video_name = file_no_ext.split(sep)[-1]
+        return video_name
+
+    def get_frames_range(self, idx):
+        """
+        Converts a flattened representation of the indices into the indices of the frames in the original video
+        """
+        video_idx, clip_idx = self.get_clip_location(idx)
+        start_frame = clip_idx * self.step
```

Comment:
I don't think this is correct. `clip_idx` doesn't necessarily need to be uniform, so you can have for specific codecs things like `[0, 1, 2, 4, 5, 6]` which makes this not valid.

Reply:
Doesnt clip idx mean the index of the clip in a certain video?
For my experiments, I assumed it indexes each clip, e.g. clips would be numbered by the order they are extracted from the video.

Reply:
`clip_idx` is the index of a particular clip. But each clip can have non-consecutive elements.
For example, take `video_pts = [0, 1, 3, 4, 5]`, with a `step` of 2 and `clip_length=3`, we have:
```
clips_pts = [
    [0, 1, 3],
    [3, 4, 5]
]
```
so in this case, the start pts for clip 0 is 0, and for clip 1 is 3, although the step is 2.

Plus, we are moving away from keeping the pts as relative to the stream, and we will be using instead seconds, so the concept of `start_frame` would become a bit less clear (it would eventually be `start_frame_pts`)